src/Communication/Connection.php

```diff
@@ -106,6 +106,35 @@ public function __construct($socketClient, LoggerInterface $logger = null, int $
 
         // create socket client
         if (\is_string($socketClient)) {
+            if (\in_array(\parse_url($socketClient, \PHP_URL_SCHEME), ['http', 'https'])) {
```

Comment:
Wouldn't it be better to extract all this code into a `private/protected` method? something like `retrieveWsDebuggerUrl($endpoint string)`